proto/substrait/algebra.proto

```diff
@@ -63,6 +63,22 @@ message ReadRel {
   Expression.MaskExpression projection = 4;
   substrait.extensions.AdvancedExtension advanced_extension = 10;
 
+  message PartitionColumns {
+    message RegularColumn {}
+    message PartitionColumn {
+      Expression.Literal value = 1;
+    }
+    message ColumnType {
+      oneof column_type {
+        RegularColumn regular = 1;
+        PartitionColumn partition = 2;
+      }
+    }
+    repeated ColumnType partition_columns = 1;
+  }
+
+  PartitionColumns partition_columns = 12;
```

Comment:
How is `partition_columns` related to `base_schema`?  Should the two have the exact same size?  Is there a default if `partition_columns` is blank?

Reply:
I think for simple types, `partition_columns` and `base_schema` have the same size. But for complex types, because the subfields are also stored in `base_schema`, while `partition_columns` is decided on column, so I think their size can be different. From our side, if `partition_columns` is blank, we treat all columns as normal ones. Please comment further if any other change is needed in PR, thanks for your help!